batch/src/main/java/org/openmrs/analytics/JdbcFhirMode.java

```diff
@@ -0,0 +1,236 @@
+			// TODO: A better solution is to Use join clause to get parent table, See comments left here:
+			//  https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/issues/46#issuecomment-728979138
+			
+		}
+	}
+	
+	public static class CreateIdRanges extends PTransform<PCollection<Integer>, PCollection<KV<String, Iterable<Integer>>>> {
+		
+		private final int batchSize;
+		
+		public CreateIdRanges(int batchSize) {
+			this.batchSize = batchSize;
+		}
+		
+		@Override
+		public PCollection<KV<String, Iterable<Integer>>> expand(PCollection<Integer> ids) {
+			return ids.apply("Distribute", ParDo.of(new DoFn<Integer, KV<String, Integer>>() {
+				
+				@ProcessElement
+				public void processElement(ProcessContext c) {
+					int count = c.element();
+					int ranges = count / batchSize;
+					for (int i = 0; i < ranges; i++) {
+						int rangeFrom = i * batchSize;
+						int rangeTo = (i + 1) * batchSize;
+						String range = String.format("%s,%s", rangeFrom, rangeTo);
+						c.output(KV.of(range, rangeFrom));
+					}
+					if (count > ranges * batchSize) {
+						int rangeFrom = ranges * batchSize;
+						int rangeTo = ranges * batchSize + count % batchSize;
+						String range = String.format("%s,%s", rangeFrom, rangeTo);
+						c.output(KV.of(range, rangeFrom));
+					}
+				}
+			})).apply("Group By", GroupByKey.create()); //necessary for returning an Iterable of ranges
+		}
+	}
+	
+	public static class FetchUuIds extends PTransform<PCollection<KV<String, Iterable<Integer>>>, PCollection<String>> {
+		
+		private final String tableName;
+		
+		private final int batchSize;
+		
+		private final JdbcIO.DataSourceConfiguration dataSourceConfig;
+		
+		public FetchUuIds(String tableName, int batchSize, JdbcIO.DataSourceConfiguration dataSourceConfig) {
+			this.tableName = tableName;
+			this.batchSize = batchSize;
+			this.dataSourceConfig = dataSourceConfig;
+		}
+		
+		@Override
+		public PCollection<String> expand(PCollection<KV<String, Iterable<Integer>>> idRanges) {
+			String tableId = this.tableName + "_id";
+			return idRanges.apply(String.format("Read UUIDs from %s", this.tableName),
+			    JdbcIO.<KV<String, Iterable<Integer>>, String> readAll().withDataSourceConfiguration(this.dataSourceConfig)
+			            .withFetchSize(this.batchSize)
+			            .withParameterSetter(new JdbcIO.PreparedStatementSetter<KV<String, Iterable<Integer>>>() {
+				            
+				            @Override
+				            public void setParameters(KV<String, Iterable<Integer>> element,
+				                    PreparedStatement preparedStatement) throws Exception {
+					            String[] range = Objects.requireNonNull(element.getKey()).split(",");
+					            preparedStatement.setInt(1, Integer.parseInt(range[0]));
+					            preparedStatement.setInt(2, Integer.parseInt(range[1]));
+				            }
+			            }).withOutputParallelization(false)
```

Comment:
What is the reason for setting this `false` here?

Reply:
specific to JDBC, enables querying  the JDBC database in parallel